### ISSUE TITLE
Embedded TeX in BibTeX

### ISSUE THREAD

**Issue by ghost** [inukshuk/jekyll-scholar#2]
Are there plans in getting jekyll-scholar to handle TeX commands within the BibTex files?  For example, I could have a title that looks like:

```
title={{$I_{S}A$}},
...
```

Any suggestions on how to handle this?

**Comment by dpo** [comment]
In my field, the title of virtually every bib entry has LaTeX commands in it, going from `\textsf{...}` to math. Has anybody made progress on this issue? I have MathJax enabled on my site but yet an entry with math only and no other LaTeX commands isn't rendered properly, even if I disable the `latex` filter.

**Comment by inukshuk** [comment]
The latex filter converts only LaTeX directives which can be translated to unicode; that doesn't include directivees like `textsf` -- however, it's quite easy to add such filters yourself for jekyll-scholar. If you have a lot of math in your bibliography, I suggest to enable MathJax and add custom filters for those directives which it does not cover.

For reference, look at the [superscript](https://github.com/inukshuk/jekyll-scholar/blob/master/lib/jekyll/scholar/plugins/superscript.rb) filter that was added recently -- simply add such a filtler to your `_plugins` folder and add its name to the `bibtex_filters` array in your configuration.

**Comment by dpo** [comment]
Thanks for the pointers! Changing simple markup is easy with filters, and that works well. But for math, I'm not sure what the filter should look like. [MathJax understands LaTeX markup](http://docs.mathjax.org/en/latest/start.html#tex-and-latex-input) but after applying the `latex` filter, the dollar signs are gone, so it doesn't get a chance.

On a side note,  there is a small problem with the regexp used in the `superscript` filter. Currently, it matches the longest string between a pair of braces but I think what you want is the *shortest* string, or else, applying the filter to `E = mc\textsuperscript{2} where E is {C}apitalized` puts everything between the `2` and the `C` in superscript. It's enough to change the regex to `/\\textsuperscript(\{((?:.|\g<1>)*?)\})/` with an extra `?` after the `*`.

**Comment by inukshuk** [comment]
Yes, so to work with MathJax the idea is to *disable* the LaTeX filter; in your case you would add additional custom filters to convert everything else that's not covered by MathJax already (including conversions that the default latex filter would normally handle; you can just copy conversions from the latex-decode gem as needed) -- it's a bit sad, but we haven't figured out a better approach to this yet.

PRs are always welcome if you found a bug!

**Comment by dpo** [comment]
So here's what I did and it seems to work:
1. Enable MathJax on my site
2. clone [`latex-decode`](https://github.com/inukshuk/latex-decode) locally, and comment out https://github.com/inukshuk/latex-decode/blob/master/lib/latex/decode.rb#L41
3. in my site's `_plugins/ext.rb`, add
```
$:.unshift '/path/to/latex-decode/lib'  # susbtitute your local path
require 'jekyll/scholar'
```
to use the local version of `latex-decode`.

Now I have proper math in my bib entries!

In view of this, it would be good to have a way to enable/disable `latex-decode` filters. Simply disabling the `math` filter does the trick. And perhaps the `math` filter should be renamed `unicode-math` or something in that vein, to avoid confusion with MathJax?!

**Comment by inukshuk** [comment]
Good idea; we could add a few filters which call the latex decode modules individually; then you could swap out latex filter with any combination of the individual ones.

By the way, instead of overriding the entire gem, you can just override the decode method (or better yet the decode method of the Math module to do nothing) in your `ext.rb` -- that would probably make your code easier to handle because you don't have to clone into latex-decode.

**Comment by dpo** [comment]
Thanks. That's indeed simpler:
```ruby
require 'latex/decode'
# Disable the Math module of latex-decode as it interferes with MathJax
module LaTeX
  module Decode
    class Maths < Decoder
      def self.decode! (string)
        string
      end
    end
  end
end
```

**Comment by ed359** [comment]
I also want to keep any inline math unchanged (for mathjax to parse) but to filter other tex commands. One problem with the above solution of disabling the math parsing is that braces are still stripped out, breaking the display in some cases. My fix was to also modify `strip_braces`:
```ruby
module Base
  module_function 
  def strip_braces (string)

    original_inline_math = string.scan(/\$[^\$]*\$/)

    string.gsub!(/(^|[^\\])([\{\}]+)/, '\1')
    string.gsub!(/\\(\{|\})/, '\1')
    
    stripped_inline_math = string.scan(/\$[^\$]*\$/)

    stripped_inline_math.zip(original_inline_math).each do |stripped, original|
      string.sub!(stripped, original)
    end
    string
  end
end
```

**Comment by andreas-bulling** [comment]
@xu-cheng We have a case where there is a superscript in the paper title: `V$^2$Dial`

Does your approach also work in this case, i.e. can it be used to properly render this in the output?
